alpaca_trade_api/rest.py

```diff
@@ -236,10 +236,12 @@ def _one_request(self, method: str, url: URL, opts: dict, retry: int):
             # retry if we hit Rate Limit
             if resp.status_code in retry_codes and retry > 0:
                 raise RetryException()
-            if 'code' in resp.text:
+            if resp.headers.get('content-type') == 'application/json':
                 error = resp.json()
-                if 'code' in error:
```

Comment:
- The content type check is to avoid json unmarshaling errors
- The `message in error` check is to avoid `KeyError` in `APIError.__init__`
- This should keep the compatibility with the existing error handling